src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1013,6 +1024,20 @@ void primitive_inst::realloc_if_needed(bool prev_execution_skipped) {
             GPU_DEBUG_CODE(memalloc_info += (((_outputs.size() > 1) ? ("o" + to_string(i) + ":") : "") +
                                   (_outputs[i]->from_memory_pool ? "from_pool" : "new_alloc"));)
             GPU_DEBUG_PROFILED_STAGE_MEMALLOC_INFO(memalloc_info);
+
+            if (need_reset_output_memory() && _outputs[i]->is_mem_from_padded_pool()) {
+                bool need_reset_padded = memory_pool.update_recent_layout_from_padded_pool(actual_layouts[i],
+                                                                                    _node->get_unique_id(),
+                                                                                    get_network_id());
+                if (need_reset_padded) {
+                    GPU_DEBUG_TRACE_DETAIL << id() << " : Need reset output memory from padded pool" << std::endl;
+                    auto& stream = get_network().get_stream();
+                    if (stream.get_queue_type() == QueueTypes::out_of_order) {
+                        stream.wait_for_events({_deps[0].first->_impl_params->out_event});
```

Comment:
I think it is necessary to check the performance impact of these changes. Did you run the performance tests?